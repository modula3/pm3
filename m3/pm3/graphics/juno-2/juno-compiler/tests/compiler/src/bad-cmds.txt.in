IF VAR a,b IN a = MAX(b,2) -> SKIP END FI
IF VAR x IN x = 3 -> IF VAR y IN INT(y) -> SKIP END FI END FI
IF VAR a,b,d,e IN INT(d) AND 2=e*d AND a+6=b AND 8=a -> SKIP END FI
IF VAR w IN (E x :: x*x=w) OR (E y :: y*y=3) -> SKIP END FI
IF VAR x IN NOT NOT NOT x = 3 -> SKIP END FI
IF VAR x, y IN NOT NOT x = 2 OR y = 3 -> SKIP END FI
IF VAR x IN NOT NOT NOT (E y :: y = x) AND x = x / 2 -> SKIP END FI
IF VAR x IN x := 1, 2 END FI
IF VAR x, y IN x, y := 1 END FI
