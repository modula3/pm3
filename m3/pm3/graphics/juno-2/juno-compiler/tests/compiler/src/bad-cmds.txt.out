Each line you enter will be parsed and compiled as a total command.

> IF VAR a,b IN a = MAX(b,2) -> SKIP END FI

*** Compilation error: You can't write this in a constraint
MAX(b[2], 2)

> IF VAR x IN x = 3 -> IF VAR y IN INT(y) -> SKIP END FI END FI

*** Compilation error: You can't write this in a constraint
INT(y[2])

> IF VAR a,b,d,e IN INT(d) AND 2=e*d AND a+6=b AND 8=a -> SKIP END FI

*** Compilation error: You can't write this in a constraint
INT(d[3])

> IF VAR w IN (E x :: x*x=w) OR (E y :: y*y=3) -> SKIP END FI

*** Compilation error: You can't write this in a constraint
(E x[2] :: x[2] * x[2] = w[1]) OR (E y[3] :: y[3] * y[3] = 3)

> IF VAR x IN NOT NOT NOT x = 3 -> SKIP END FI

*** Compilation error: You can't write this in a constraint
NOT NOT NOT x[1] = 3

> IF VAR x, y IN NOT NOT x = 2 OR y = 3 -> SKIP END FI

*** Compilation error: You can't write this in a constraint
NOT NOT x[1] = 2 OR y[2] = 3

> IF VAR x IN NOT NOT NOT (E y :: y = x) AND x = x / 2 -> SKIP END FI

*** Compilation error: You can't write this in a constraint
NOT NOT NOT (E y[2] :: y[2] = x[1])

> IF VAR x IN x := 1, 2 END FI

*** Compilation error: The number of variables on the left
is different from the
number of terms on the right
x := 1, 2

> IF VAR x, y IN x, y := 1 END FI

*** Compilation error: The number of variables on the left
is different from the
number of terms on the right
x, y := 1

> ^D
